src/containers/BooksForm.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { categories } from '../constants';
+import { createBook } from '../actions';
+
+const mapDispatchToProps = dispatch => ({
+  createBook: book => dispatch(createBook(book)),
+});
+
+const BooksForm = ({
+  createBook,
+}) => {
+  let state;
+
+  const initialState = {
+    title: '',
+    category: categories[0],
+  };
+
+  const setState = newState => {
+    state = {
+      ...state,
+      ...newState,
+    };
+  };
```

Comment:
- [ ] We should let React manipulate the state. In order to do that, you can make this component to be a stateful component or use [React useState](reactjs.org/docs/hooks-state.html).

- [ ] We should not also try to manipulate the DOM directly with the `document`. This is due to the fact that React interacts with the virtual DOM instead of the real DOM. Manipulating the real DOM directly can lead to unexpected errors. And you also do not get the React performance benefits when you do so. Instead of that, we should use the React approach to build this form. You can use this [article here](https://www.telerik.com/blogs/how-to-build-forms-with-react-the-easy-way) to see an example. 

If you feel like you really need to do an operation that required to use the document in React, there is a safe way to do it, and it is through the use of [React useRef](reactjs.org/docs/hooks-reference.html#useref).